src_c/surface.c

```diff
@@ -2029,13 +2041,19 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
         if (doreturn) {
             retrect = NULL;
             retrect = pgRect_New(&dest_rect);
-            if (PyList_Append(ret, retrect) != 0) {
-                bliterrornum = BLITS_ERR_PY_EXCEPTION_RAISED;
-                goto bliterror;
+            if (issequence) {
+                PyList_SET_ITEM(ret, curriter, retrect);
+            }
+            else {
+                if (PyList_Append(ret, retrect) != 0) {
+                    bliterrornum = BLITS_ERR_PY_EXCEPTION_RAISED;
+                    goto bliterror;
```

Comment:
This could be collapsed into a single `else if`, not sure if that would be better. What do you think?

Reply:
This is probably the cleanes way, don't even need an else. Alternative would be this
```C
else if (PyList_Append(ret, retrect)) {
      Py_DECREF(retrect);
  }
else {
    Py_DECREF(retrect);
    retrect = NULL;
    bliterrornum = BLITS_ERR_PY_EXCEPTION_RAISED;
    goto bliterror;
}
retrect = NULL; /* Clear to avoid double deref on errors *
}
```

Reply:
which is even more convoluted

Reply:
In the end i made the change anyway.